packages/react-tooltip/src/components/Tooltip/Tooltip.test.tsx

```diff
@@ -42,13 +42,17 @@ describe('Tooltip', () => {
   it('renders the content of a nontrivial tooltip', () => {
     const result = render(
       <Tooltip
-        id="the-tooltip-id"
-        data-testid="the-tooltip"
-        content={
-          <span>
-            This is a <strong>formatted</strong> tooltip
-          </span>
-        }
+        content={{
+          children: (
+            <span>
+              This is a <strong>formatted</strong> tooltip
+            </span>
+          ),
+          id: 'the-tooltip-id',
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          'data-testid': 'the-tooltip',
```

Comment:
As our typings are strict we don't allow to pass `data-` attributes:

```
src/components/Tooltip/Tooltip.test.tsx:52:11 - error TS2322: Type '{ children: JSX.Element; id: string; 'data-testid': string; }' is not assignable to type 'ShorthandProps<{ as?: "div" | undefined; } & Pick<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, "key" | keyof HTMLAttributes<...>> & { ...; } & { ...; }>'.
  Object literal may only specify known properties, and ''data-testid'' does not exist in type 'ReactElement<any, string | JSXElementConstructor<any>> | ReactNodeArray | ReactPortal | ({ as?: "div" | undefined; } & Pick<...> & { ...; } & { ...; })'.

52           'data-testid': 'the-tooltip',
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This probably should be turned into an issue.

Reply:
FYI, I'm removing the data attributes from the tests with this PR: https://github.com/microsoft/fluentui/pull/21331.

I agree it's still an issue that our types don't allow for data attributes, but at least it won't be an issue in this particular case.